src/main/kotlin/io/github/gradlenexus/publishplugin/InitializeNexusStagingRepository.kt

```diff
@@ -42,16 +44,83 @@ open class InitializeNexusStagingRepository @Inject constructor(
         val stagingProfileId = determineStagingProfileId(repository, client)
-        logger.info("Creating staging repository for {} at {}, stagingProfileId '{}'", repository.name, serverUrl, stagingProfileId)
-        val descriptor = client.createStagingRepository(stagingProfileId, repositoryDescription.get())
-        val consumerUrl = HttpUrl.get(serverUrl)!!.newBuilder().addEncodedPathSegments("repositories/${descriptor.stagingRepositoryId}/content/").build()
-        logger.lifecycle("Created staging repository '{}' at {}", descriptor.stagingRepositoryId, consumerUrl)
+
+        var descriptor: StagingRepositoryDescriptor? =
+            when (val repositorySearchMode = repositorySearchMode.get()) {
+                RepositorySearchMode.CREATE -> null /* handled below */
+                RepositorySearchMode.FIND_OR_CREATE,
+                RepositorySearchMode.FIND_OR_FAIL,
+                RepositorySearchMode.FIND_OR_NULL -> {
+                    val descriptionRegex = repositorySearchRegex.get()
+                    logger.info(
+                        "Searching for an existing staging repository for {} at {}, stagingProfileId '{}' via description regular expression {}",
+                        repository.name,
+                        serverUrl,
+                        stagingProfileId,
+                        descriptionRegex
+                    )
+                    try {
+                        val descriptor = client.findStagingRepository(stagingProfileId, Regex(descriptionRegex))
+                        logger.lifecycle(
+                            "Staging repository for {} at {}, stagingProfileId '{}', descriptionRegex '{}' is '{}'",
+                            repository.name,
+                            serverUrl,
+                            stagingProfileId,
+                            descriptionRegex,
+                            descriptor.stagingRepositoryId
+                        )
+                        descriptor
+                    } catch (e: NoSuchElementException) {
+                        when (repositorySearchMode) {
+                            RepositorySearchMode.FIND_OR_FAIL -> throw e
+                            RepositorySearchMode.FIND_OR_NULL -> return
+                            else -> {
+                                logger.info(
+                                    "No staging repository found for {} at {}, stagingProfileId '{}', descriptionRegex '{}'",
+                                    repository.name,
+                                    serverUrl,
+                                    stagingProfileId,
+                                    descriptionRegex
+                                )
+                                null
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
Sorry, but that is too complex as a "one-liner" assigning a value to `descriptor`. 5 levels of nested statement and clumsy conditional logic. It would need to be refactored into smaller pieces, however, please take a look at the "general comment" to new approach before doing any actions here.